src/main/java/org/jetbrains/bsp/bazel/server/resolvers/QueryResolver.java

```diff
@@ -2,26 +2,21 @@
 
 import com.google.devtools.build.lib.query2.proto.proto2api.Build;
 import java.io.IOException;
-import java.util.concurrent.Semaphore;
+import org.jetbrains.bsp.bazel.server.data.ProcessResults;
 
 public class QueryResolver {
 
-  private final ProcessResolver processResolver;
-  private final Semaphore processLock;
+  private final BazelRunner bazelRunner;
```

Comment:
why did u remove a semaphore?

Reply:
because locking and releasing it is now a responsibility of the ProcessResolver

Reply:
did u mean `BazelRunner`?

Reply:
yup